### PR TITLE
1/x Encourage admin users to connect to a database

### PR DESCRIPTION
**Pull request by WiNloSt**
Epic: https://github.com/metabase/metabase/issues/31015
[Product doc](https://www.notion.so/metabase/Really-encourage-admins-to-add-a-DB-9c8c3a7d08ca45cb8672a154220f443e?pvs=4)

### Description

This PR adds a banner on top of the app bar when these conditions are met
1. User is an admin
2. A Metabase instance is on a paid plan
3. The instance only has a sample database.

There is another variation of the banner that we will not show `Connect your database` button when users is on `/admin/databases/create` otherwise, this button should be visible at all time (given all 3 conditions are met still).

### How to verify

Try to play around with the combinations of the above conditions, only when the 3 conditions are met then the banner would show up.

#### App
image

#### Admin add database page
image

#### Other admin pages
image

### Checklist

- [x] Tests have been added/updated to cover changes in this PR

### CODE REVIEW

**Review by kdoh — APPROVED** [review]
Looking good to me. 👍

File: frontend/src/metabase/nav/components/DatabasePromptBanner/DatabasePromptBanner.styled.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+const VIBRANT_BLUE = "#1888EC";
+
+export const DatabasePromptBannerRoot = styled.div`
+  background: ${VIBRANT_BLUE};
+  color: ${color("white")};
+  display: flex;
+  align-items: center;
+  flex-wrap: wrap;
+
+  ${breakpointMinSmall} {
```
Comment (WiNloSt) [inline frontend/src/metabase/nav/components/DatabasePromptBanner/DatabasePromptBanner.styled.ts:18]: Add breakpoints to various elements to make the mobile view looks more polished. [The design is here](https://www.figma.com/file/H8NL31kXy2bNxuXtSKX0Fu/Really-encourage-admins-to-add-a-DB?type=design&node-id=27-92&t=ktyrZcQNXHgbNUrR-0).

#### Before
- image
- image

#### After
- image
- image

File: frontend/src/metabase/components/AppBanner/AppBanner.tsx [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+import { getUserIsAdmin } from "metabase/selectors/user";
+import { PaymentBanner } from "metabase/nav/components/PaymentBanner/PaymentBanner";
+import { DatabasePromptBanner } from "metabase/nav/components/DatabasePromptBanner";
+
+export const AppBanner = () => {
+  const isAdmin = useSelector(getUserIsAdmin);
+  const tokenStatusStatus = useSelector(
+    state => getSetting(state, "token-status")?.status,
+  );
+  if (PaymentBanner.shouldRender({ isAdmin, tokenStatusStatus })) {
```
Comment (WiNloSt) [inline frontend/src/metabase/components/AppBanner/AppBanner.tsx:12]: Only render one banner at a time, and if somehow both banner conditions are met, only render the payment banner. [Discussion here.](https://metaboat.slack.com/archives/C057T1QTB3L/p1685549180307149?thread_ts=1685547913.580079&cid=C057T1QTB3L)

File: frontend/src/metabase/nav/components/DatabasePromptBanner/DatabasePromptBanner.tsx [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+  DatabasePromptBannerRoot,
+  Prompt,
+  GetHelpButton,
+} from "./DatabasePromptBanner.styled";
+
+interface DatabasePromptBannerProps {
+  location: Location;
+}
+
+export const DatabasePromptBanner = withRouter(
```
Comment (ranquild) [inline frontend/src/metabase/nav/components/DatabasePromptBanner/DatabasePromptBanner.tsx:23]: HoCs are discouraged, could you pass `Location` from the nearest component rendered by the router instead?

File: frontend/src/metabase/nav/components/DatabasePromptBanner/DatabasePromptBanner.tsx [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+  function DatabasePromptBannerView({ location }: DatabasePromptBannerProps) {
+    const isAdmin = useSelector(getUserIsAdmin);
+    const isPaidPlan = useSelector(getIsPaidPlan);
+    const { data: databases } = useDatabaseListQuery({
```
Comment (ranquild) [inline frontend/src/metabase/nav/components/DatabasePromptBanner/DatabasePromptBanner.tsx:27]: nit `data: databases = []` if you want to default an empty list

File: frontend/src/metabase/nav/components/PaymentBanner/PaymentBanner.tsx [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+          </ExternalLink>
+        )} to restore Pro functionality.`}
+      </Banner>
+    );
+  }
+
+  return null;
+};
+
+PaymentBanner.shouldRender = ({
```
Comment (ranquild) [inline frontend/src/metabase/nav/components/PaymentBanner/PaymentBanner.tsx:48]: Attaching functions to components isn't something common on the frontend, could you please make this a regular function instead?

**Review by ranquild — COMMENTED** [review]
@WiNloSt could you add one E2E test that verifies you can see the new banner, click on the button and see the form to add a database? Otherwise the PR looks good!

File: frontend/src/metabase/nav/components/DatabasePromptBanner/DatabasePromptBanner.tsx [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+  }
+
+  const isOnAdminAddDatabasePage = location.pathname.startsWith(
+    "/admin/databases/create",
+  );
+
+  return (
+    <DatabasePromptBannerRoot
+      role="banner"
+      aria-label={t`Database prompt banner`}
```
Comment (WiNloSt) [inline frontend/src/metabase/nav/components/DatabasePromptBanner/DatabasePromptBanner.tsx:44]: @mazameli @ranquild I'm not sure this warrant a new translation. The label is only used for the E2E test, (and hopefully for screen readers too).

Please feel free to suggest a better name if it should be translated.

File: frontend/src/metabase/nav/components/DatabasePromptBanner/DatabasePromptBanner.tsx [hunk 7]
```diff
@@ -0,0 +1,59 @@
+import { t } from "ttag";
```
Comment (WiNloSt) [inline frontend/src/metabase/nav/components/DatabasePromptBanner/DatabasePromptBanner.tsx:1]: @ranquild I added one test that test a happy path of this feature, the rest should already be covered with the integration test.

File: frontend/src/metabase/nav/components/DatabasePromptBanner/DatabasePromptBanner.tsx (hunk 6 above)
Comment (ranquild) [reply frontend/src/metabase/nav/components/DatabasePromptBanner/DatabasePromptBanner.tsx:44]: I wouldn't add an aria-label here, at least not normally. We can either add `data-testid` or just check for something in the banner.

Comment (WiNloSt) [reply frontend/src/metabase/nav/components/DatabasePromptBanner/DatabasePromptBanner.tsx]: Removed, and use `cy.findAllByRole("banner").first()` to target this banner instead.

File: frontend/src/metabase/components/AppBanner/AppBanner.jsx
Comment (WiNloSt) [inline frontend/src/metabase/components/AppBanner/AppBanner.jsx:1]: The file changed to TypeScript. And move the content to `PaymentBanner` instead.

File: frontend/src/metabase/components/AppBanner/AppBanner.tsx [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+interface AppBannerProps {
+  location: Location;
+}
+
+export const AppBanner = ({ location }: AppBannerProps) => {
+  const isAdmin = useSelector(getUserIsAdmin);
+  const tokenStatusStatus = useSelector(
+    state => getSetting(state, "token-status")?.status,
+  );
+  if (shouldRenderPaymentBanner({ isAdmin, tokenStatusStatus })) {
```
Comment (WiNloSt) [inline frontend/src/metabase/components/AppBanner/AppBanner.tsx:21]: Only render one banner at a time, and if somehow both banner conditions are met, only render the payment banner. [Discussion here.](https://metaboat.slack.com/archives/C057T1QTB3L/p1685549180307149?thread_ts=1685547913.580079&cid=C057T1QTB3L)

File: frontend/src/metabase/dashboard/components/Dashboard/Dashboard.jsx [hunk 9]
```diff
@@ -332,7 +331,9 @@ class Dashboard extends Component {
             <DashboardBody isEditingOrSharing={isEditing || isSharing}>
               <ParametersAndCardsContainer
                 data-testid="dashboard-parameters-and-cards"
-                ref={element => (this.parametersAndCardsContainerRef = element)}
+                shouldMakeDashboardHeaderStickyAfterScrolling={
```
Comment (WiNloSt) [inline frontend/src/metabase/dashboard/components/Dashboard/Dashboard.jsx:334]: This prop controls whether we will render this container with `overflow: auto` or not
image

If we want the dashboard header to be sticky as well. We need to set `overflow-y: auto`

Here are some variations of the UI

1. Long dashboard with more info panel open
image
2. Long dashboard with subscription sidebar open
image
3. Editing dashboard
image

4. Full-screen dashboards
Screenshot 2023-06-07 at 3 09 48 PM


This follows the current behaviors on the dashboard, I think overall the whole UI should be standardized since you could probably see that we scroll on different containers and make different UIs sticky as well.

File: frontend/src/metabase/dashboard/components/Dashboard/Dashboard.styled.jsx [hunk 10]
```diff
@@ -87,7 +80,12 @@ export const HeaderContainer = styled.header`
 
 export const ParametersAndCardsContainer = styled.div`
   flex: auto;
+  overflow-y: ${({ shouldMakeDashboardHeaderStickyAfterScrolling }) =>
+    shouldMakeDashboardHeaderStickyAfterScrolling ? "auto" : "visible"};
   overflow-x: hidden;
+  @supports (overflow-x: clip) {
+    overflow-x: clip;
```
Comment (WiNloSt) [inline frontend/src/metabase/dashboard/components/Dashboard/Dashboard.styled.jsx:87]: For the rest [10% of browsers](https://caniuse.com/?search=overflow%3Aclip) that don't support `overflow-x: clip`, we just don't make the parameter sticky at all (please note that in different layouts where we don't hide the dashboard header, the parameters will still be sticky together with the dashboard header). [Bruno is okay with this](https://metaboat.slack.com/archives/C505ZNNH4/p1686054411768389?thread_ts=1686054215.543739&cid=C505ZNNH4).

#### 90% of browsers
image

#### 10% of browsers
image

File: frontend/src/metabase/dashboard/components/Dashboard/stickyParameters.js [hunk 11]
```diff
@@ -13,8 +13,6 @@ export const updateParametersWidgetStickiness = dashboard => {
   );
 
   if (shouldToggleStickiness) {
-    updateParametersAndCardsContainerStyle(dashboard, shouldBeSticky);
```
Comment (WiNloSt) [inline frontend/src/metabase/dashboard/components/Dashboard/stickyParameters.js:16]: This isn't needed because it's handled by `position: sticky`

File: frontend/src/metabase/dashboard/components/Dashboard/stickyParameters.unit.spec.js [hunk 12]
```diff
@@ -15,7 +15,6 @@ describe("updateParametersWidgetStickiness", () => {
 
     const dashboard = {
       parametersWidgetRef: { offsetTop },
-      parametersAndCardsContainerRef: { style: {} },
```
Comment (WiNloSt) [inline frontend/src/metabase/dashboard/components/Dashboard/stickyParameters.unit.spec.js:18]: update tests since we don't use this anymore. This was used when we weren't using `position: sticky` and rely on JS positional calculations.

File: frontend/src/metabase/nav/components/PaymentBanner/PaymentBanner.tsx [hunk 13]
```diff
@@ -0,0 +1,54 @@
+import { jt, t } from "ttag";
```
Comment (WiNloSt) [inline frontend/src/metabase/nav/components/PaymentBanner/PaymentBanner.tsx:1]: This should look mostly like the previous version of the `AppBanner`
https://github.com/metabase/metabase/blob/fdc07959b9369547221ac5ab897a951f24160f46/frontend/src/metabase/components/AppBanner/AppBanner.jsx

The only notable change is the logic to determine whether we should display this banner is lifted up, so we can decide when to display this banner or the database prompt banner.

File: e2e/test/scenarios/dashboard/dashboard.cy.spec.js [hunk 14]
```diff
@@ -693,7 +693,9 @@ function checkOptionsForFilter(filter) {
     .and("not.contain", "Dashboard filters");
 
   // Get rid of the open popover to be able to select another filter
-  cy.findByText("Pick one or more filters to update").click();
+  // Uses force: true because the popover is covering this text. This happens
+  // after we introduce the database prompt banner.
+  cy.findByText("Pick one or more filters to update").click({ force: true });
```
Comment (ranquild) [inline e2e/test/scenarios/dashboard/dashboard.cy.spec.js:698]: Can we scroll to it somehow? e.g. scrollIntoView or similar

File: frontend/src/metabase/dashboard/components/Dashboard/Dashboard.styled.jsx [hunk 15]
```diff
@@ -87,7 +80,12 @@ export const HeaderContainer = styled.header`
 
 export const ParametersAndCardsContainer = styled.div`
   flex: auto;
+  overflow-y: ${({ shouldMakeDashboardHeaderStickyAfterScrolling }) =>
```
Comment (ranquild) [inline frontend/src/metabase/dashboard/components/Dashboard/Dashboard.styled.jsx:83]: nit: shorter parameter name? :)

File: frontend/src/metabase/components/AppBanner/AppBanner.tsx [hunk 16]
```diff
@@ -0,0 +1,28 @@
...
+  PaymentBanner,
+  shouldRenderPaymentBanner,
+} from "metabase/nav/components/PaymentBanner/PaymentBanner";
+import { DatabasePromptBanner } from "metabase/nav/components/DatabasePromptBanner";
+
+interface AppBannerProps {
+  location: Location;
+}
+
+export const AppBanner = ({ location }: AppBannerProps) => {
```
Comment (ranquild) [inline frontend/src/metabase/components/AppBanner/AppBanner.tsx:16]: I don't see tests for this component. Do they exist?

File: frontend/src/metabase/nav/components/PaymentBanner/PaymentBanner.tsx [hunk 17]
```diff
@@ -0,0 +1,54 @@
...
+import Banner from "metabase/components/Banner";
+import ExternalLink from "metabase/core/components/ExternalLink";
+import MetabaseSettings from "metabase/lib/settings";
+
+interface PaymentBannerProps {
+  isAdmin: boolean;
+  tokenStatusStatus: string | undefined;
+}
+
+export const PaymentBanner = ({
```
Comment (ranquild) [inline frontend/src/metabase/nav/components/PaymentBanner/PaymentBanner.tsx:11]: I don't see tests for this component. Do they exist?

File: frontend/src/metabase/selectors/settings.ts [hunk 18]
```diff
@@ -40,3 +40,12 @@ const getUtmSource = (features: TokenFeatures) => {
     return features.hosting ? "starter" : "oss";
   }
 };
+
+const getTokenFeatures = (state: State) => getSetting(state, "token-features");
+
+export const getIsPaidPlan = createSelector(
```
Comment (ranquild) [inline frontend/src/metabase/selectors/settings.ts:46]: I don't see tests for this utility. Do they exist?

**Review by ranquild — COMMENTED** [review]
Looks good, comments about testing only

File: e2e/test/scenarios/dashboard/dashboard.cy.spec.js (hunk 14 above)
Comment (WiNloSt) [reply e2e/test/scenarios/dashboard/dashboard.cy.spec.js:698]: This is the place that this test is clicking
Screenshot 2023-06-08 at 4 36 09 PM

Unfortunately, this is a full-height layout. So it's impossible to scroll anywhere on this page for this particular setup. The problem is when clicking the **Category** filter. It will look like the following screenshot which completely cover the text element we want to click.
image

File: frontend/src/metabase/dashboard/components/Dashboard/Dashboard.styled.jsx (hunk 15 above)
Comment (WiNloSt) [reply frontend/src/metabase/dashboard/components/Dashboard/Dashboard.styled.jsx:83]: The other approach in this file is when we specify what we pass instead. e.g. `isEditing or `isEditingOrSharing` which won't work with what we're passing here. `!isFullscreen && (isEditing || isSharing)` so I think it's better to document what we want this flag to mean.

I'm open to new names. But this is the most concise version I could come up with at the time. I think the weirdness with the name would be getting away once we start making dashboard UI header + sticky components more consistent. You can see [in my other comment](https://github.com/metabase/metabase/pull/31210#discussion_r1221121046) that we have quite different variations for dashboard headers. Notably different elements will be sticky depending on whether we're opening a subscription sidebar or the info sidebar.

File: frontend/src/metabase/components/AppBanner/AppBanner.tsx (hunk 16 above)
Comment (WiNloSt) [reply frontend/src/metabase/components/AppBanner/AppBanner.tsx:16]: They don't exist. This component used to house what's in the `PaymentBanner` which is newly introduced in this PR. The only tests for the previously `AppBanner` aka `PaymentBanner` (now) are [E2E tests](https://github.com/metabase/metabase/blob/37a24a471908b2caf64698418614a905a80e4d66/e2e/test/scenarios/admin/subscription/payment-failure.cy.spec.js#L11).

Would it make sense to add a few assertions in that file `e2e/test/scenarios/admin/subscription/payment-failure.cy.spec.js` to assert that the database prompt banner won't be displayed, because the database prompt banner will only display when the payment banner isn't showing?

File: frontend/src/metabase/nav/components/PaymentBanner/PaymentBanner.tsx (hunk 17 above)
Comment (WiNloSt) [reply frontend/src/metabase/nav/components/PaymentBanner/PaymentBanner.tsx:11]: They exist. Here are the tests https://github.com/metabase/metabase/blob/master/e2e/test/scenarios/admin/subscription/payment-failure.cy.spec.js.

[Here's another related comment](https://github.com/metabase/metabase/pull/31210#discussion_r1222738533), since this component used to be named `AppBanner`

File: frontend/src/metabase/selectors/settings.ts (hunk 18 above)
Comment (WiNloSt) [reply frontend/src/metabase/selectors/settings.ts:46]: Good catch. They don't, since I just created this selector in this PR. I will add one.

Comment (WiNloSt) [reply frontend/src/metabase/selectors/settings.ts:46]: Added tests

File: frontend/src/metabase/components/AppBanner/AppBanner.tsx (hunk 16 above)
Comment (WiNloSt) [reply frontend/src/metabase/components/AppBanner/AppBanner.tsx:16]: Added tests

File: frontend/src/metabase/nav/components/PaymentBanner/PaymentBanner.tsx (hunk 17 above)
Comment (WiNloSt) [reply frontend/src/metabase/nav/components/PaymentBanner/PaymentBanner.tsx:11]: Added tests

**Comment by WiNloSt** [comment]
@ranquild I added 3 more unit test files. please see this commit https://github.com/metabase/metabase/pull/31210/commits/19cc26b0be3c11a28d2fae37163d41a7662afff2

Those tests should cover every case possible.

File: e2e/test/scenarios/admin/datamodel/editor.cy.spec.js [hunk 19]
```diff
@@ -633,7 +633,7 @@ const getFieldSection = fieldName => {
 };
 
 const moveField = (fieldIndex, deltaY) => {
-  cy.get(".Grabber").eq(fieldIndex).trigger("mousedown", 0, 0);
+  cy.get(".Grabber").eq(fieldIndex).trigger("mousedown", 0, 0, { force: true });
```
Comment (WiNloSt) [inline e2e/test/scenarios/admin/datamodel/editor.cy.spec.js:636]: Interestingly,[ this just fails](https://www.deploysentinel.com/ci/runs/6481ca4014a47e6dd93d5a65) on the latest commit https://github.com/metabase/metabase/pull/31210/commits/19cc26b0be3c11a28d2fae37163d41a7662afff2, which only added unit tests 🤔 . Maybe it has something to do with rebasing off `master` too.

But I think this is justified since other place that `trigger("mousedown", xxx)` all uses `force: true`
Screenshot 2023-06-08 at 9 24 43 PM

File: frontend/src/metabase/components/AppBanner/AppBanner.unit.spec.tsx [hunk 20]
```diff
@@ -0,0 +1,212 @@
...
+      renderingBanner: "Past-due banner",
+    },
+    {
+      shouldRenderPastDueBanner: false,
+      shouldRenderUnpaidBanner: true,
+      shouldRenderDatabasePromptBanner: true,
+      renderingBanner: "Unpaid banner",
+    },
+  ] as const)(
+    "should render $renderingBanner when should render past-due banner: $shouldRenderPastDueBanner, should render unpaid banner: $shouldRenderUnpaidBanner, should render database prompt banner: $shouldRenderDatabasePromptBanner",
```
Comment (ranquild) [inline frontend/src/metabase/components/AppBanner/AppBanner.unit.spec.tsx:121]: nit: I might be nit-picking but I don't follow `shouldDoXXX` tests. I think what we actually want to test is different combinations of token features and the `isAdmin` flag, and that you get expected results there. Now the test looks like if `shouldRenderUnpaidBanner` is true, we render the unpaid banner. That's not what we really want to test. Should be fairly straightforward change if you remove the `should` flags and set the assigned properties directly.

File: frontend/src/metabase/components/AppBanner/AppBanner.unit.spec.tsx [hunk 21]
```diff
@@ -0,0 +1,212 @@
...
+});
+
+async function assertBanner(
+  renderingBanner:
+    | "Nothing"
+    | "Past-due banner"
+    | "Unpaid banner"
+    | "Database prompt banner",
+) {
+  if (renderingBanner === "Nothing") {
```
Comment (ranquild) [inline frontend/src/metabase/components/AppBanner/AppBanner.unit.spec.tsx:146]: nit: You can get rid of this method if you split the `it.each` clause into 3 different clauses. Currently this is a bit hard to follow

File: frontend/src/metabase/nav/components/DatabasePromptBanner/DatabasePromptBanner.unit.spec.tsx [hunk 22]
```diff
@@ -0,0 +1,234 @@
...
+      initialRoute: isOnAdminAddDatabasePage ? "/admin/databases/create" : "/",
+      storeInitialState: state,
+      withRouter: true,
+    },
+  );
+}
+
+describe("DatabasePromptBanner", () => {
+  it("should not render for non-admin users without paid plan without connected databases", () => {
+    setup({
```
Comment (ranquild) [inline frontend/src/metabase/nav/components/DatabasePromptBanner/DatabasePromptBanner.unit.spec.tsx:71]: In contract to the previous comment, I really like these tests from the `setup` point of view - if you're an admin, on a paid plan, and have other databases and not just the sample one, you shouldn't see the paid banner. Makes sense from the product point of view.

File: frontend/src/metabase/nav/components/PaymentBanner/PaymentBanner.unit.spec.tsx [hunk 23]
```diff
@@ -0,0 +1,126 @@
...
+        isAdmin: true,
+        tokenStatusStatus: "something-else",
+      });
+
+      expectBannerToBeHidden();
+    });
+  });
+
+  describe("non-admin users", () => {
+    it.each([
```
Comment (ranquild) [inline frontend/src/metabase/nav/components/PaymentBanner/PaymentBanner.unit.spec.tsx:56]: Also good tests

File: frontend/src/metabase/nav/components/PaymentBanner/PaymentBanner.unit.spec.tsx [hunk 24]
```diff
@@ -0,0 +1,126 @@
...
+      },
+      {
+        isAdmin: false,
+        tokenStatusStatus: "unpaid",
+        shouldRenderPaymentBanner: false,
+      },
+      {
+        isAdmin: false,
+        tokenStatusStatus: "something-else",
+        shouldRenderPaymentBanner: false,
```
Comment (ranquild) [inline frontend/src/metabase/nav/components/PaymentBanner/PaymentBanner.unit.spec.tsx:106]: I'd call it something different to avoid the collision, e.g. `hasBanner` or something like this

File: frontend/src/metabase/selectors/selectors.unit.spec.ts [hunk 25]
```diff
@@ -43,3 +43,35 @@ describe("getUpgradeUrl", () => {
     expect(url.search).toEqual(params);
   });
 });
+
+describe("getIsPaidPlan", () => {
```
Comment (ranquild) [inline frontend/src/metabase/selectors/selectors.unit.spec.ts:47]: good tests

File: frontend/src/metabase/components/AppBanner/AppBanner.unit.spec.tsx (hunk 20 above)
Comment (ranquild) [reply frontend/src/metabase/components/AppBanner/AppBanner.unit.spec.tsx:121]: Other tests are implemented in the way I'd expect, if we can fix this test suite it would be great

**Review by ranquild — APPROVED** [review]
Looks good! Left some comments about one of the test suites, please take a look

File: frontend/src/metabase/components/AppBanner/AppBanner.unit.spec.tsx (hunk 20 above)
Comment (WiNloSt) [reply frontend/src/metabase/components/AppBanner/AppBanner.unit.spec.tsx:121]: I was thinking about that too. But since there are tests that cover some combinations of those flags in `DatabasePromptBanner.unit.spec.tsx` and `PaymentBanner.unit.spec.tsx` it feels as if we're repeating a lot of that code here.

To be fair, I was also thinking about removing those 2 test suites and testing every combination here in `AppBanner.unit.spec.tsx` instead. Otherwise, we will just create a superset of tests from those 2 other test suites.

What do you think?

File: frontend/src/metabase/components/AppBanner/AppBanner.tsx [hunk 26]
```diff
@@ -0,0 +1,28 @@
...
+  const isAdmin = useSelector(getUserIsAdmin);
+  const tokenStatusStatus = useSelector(
+    state => getSetting(state, "token-status")?.status,
+  );
+  if (shouldRenderPaymentBanner({ isAdmin, tokenStatusStatus })) {
+    return (
+      <PaymentBanner isAdmin={isAdmin} tokenStatusStatus={tokenStatusStatus} />
+    );
+  } else {
```
Comment (uladzimirdev) [inline frontend/src/metabase/components/AppBanner/AppBanner.tsx:25]: you don't need `else` here

Comment (WiNloSt) [reply frontend/src/metabase/components/AppBanner/AppBanner.tsx:25]: Ah yep that'll be cleaner 👍

**Review by cdeweyx — APPROVED** [review]
Works as expected, approved pending additional case handling

File: frontend/src/metabase/components/AppBanner/AppBanner.tsx (hunk 26 above)
Comment (WiNloSt) [reply frontend/src/metabase/components/AppBanner/AppBanner.tsx:25]: fixed

File: frontend/src/metabase/nav/components/PaymentBanner/PaymentBanner.unit.spec.tsx (hunk 24 above)
Comment (WiNloSt) [reply frontend/src/metabase/nav/components/PaymentBanner/PaymentBanner.unit.spec.tsx:106]: Fixed

File: frontend/src/metabase/components/AppBanner/AppBanner.unit.spec.tsx (hunk 21 above)
Comment (WiNloSt) [reply frontend/src/metabase/components/AppBanner/AppBanner.unit.spec.tsx:146]: I agree this is too hard to follow.
